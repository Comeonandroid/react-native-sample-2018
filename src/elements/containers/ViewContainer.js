import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import { View } from '../index'

const ViewContainer = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

ViewContainer.propTypes = {
  children: PropTypes.node
}

const Element = styled(View) `
  padding: 10px 15px;
`

export default ViewContainer