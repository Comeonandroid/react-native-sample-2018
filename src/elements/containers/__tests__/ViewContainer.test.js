import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import ViewContainer from '../ViewContainer'

describe('<ViewContainer />', () => {
  itBehavesLike('aThemedComponent', { subject: <ViewContainer /> })
  itBehavesLike('aComponentWithChildren', { subject: <ViewContainer /> })
})
