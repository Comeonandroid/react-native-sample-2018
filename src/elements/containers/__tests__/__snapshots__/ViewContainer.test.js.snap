// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewContainer /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<ViewContainer /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<ViewContainer /> renders component when passed in 1`] = `
<Styled(View)>
  <Component />
</Styled(View)>
`;

exports[`<ViewContainer /> renders text when passed in 1`] = `
<Styled(View)>
  Test
</Styled(View)>
`;
